oop-3-inheritance_nd_visibility/index.md

```diff
@@ -0,0 +1,454 @@
+        this.name = name;
+    }
+
+
+    public String getCategory() {
+        return category;
+    }
+
+
+    public void setCategory(String category) {
+        this.category = category;
+    }
+
+
+    @Override
+    public String toString() {
+        return "Product{" +
+               "price=" + price +
+               ", name='" + name + '\'' +
+               ", category='" + category + '\'' +
+               '}' + "I'm also a a laptop btw!";
+    }
+
+```
+
+### new solution (nice!)
+
+Why don't we use inheritance?
+
+```java
+package com.redi.j2;
+
+public class laptop extends Product {
+
+    public laptop() {
+        setCategory("electronics");
+    }
+
+
+    @Override
+    public void setPrice(float price) {
+        if (price < 100) {
+            System.out.println("I can't be THAT cheap!!");
+        }
+        else {
+            super.setPrice(price);
+        }
+    }
+
+
+    @Override
+    public String toString() {
+        return super.toString() + "I am also a laptop btw!";
+    }
+}
+```
+
+### Think of inheritance as an "is-a" relationship
+
+In Java, inheritance is an is-a relationship. That is, we use inheritance only if there exists an is-a relationship between two classes. For example,
+
+Laptop is a Prouct Car is a Vehicle Orange is a Fruit
+
+## Why do we need it?
+
+- The most important use of inheritance in Java is code reusability. The code that is present in the parent class can be directly used by the child
+  class.
+- Method overriding is also known as runtime polymorphism. Hence, we can achieve Polymorphism in Java with the help of inheritance.
+
+## Have we seen it before?
+
+**YES!!** Since day one actually. Remember this example from the very first lesson?
+
+```java
+package com.redi.j2;
+
+public class Main {
+
+    public static void main(String[] args) {
+        System.out.println("Hello World");
+
+        String name = "Memet";
+        String profession = "skydiver";
+        int age = 40;
+        boolean smoking = false;
+
+        coolOrNotCool(profession);
+
+        for (int i = 0; i < 10; i++) {
+            System.out.println("hey " + i);
+        }
+    }
+
+
+    private static void coolOrNotCool(String profession) {
+        if (profession.equals("skydiver")) {
+            System.out.println("Cool");
+        }
+        else {
+            System.out.println("Not cool");
+        }
+    }
+}
+```
+
+let's take a moment and understand what is a [`String`](https://docs.oracle.com/javase/7/docs/api/java/lang/String.html) really?
+
+## Visibility and access modifiers
+
+### Default Access Modifier - No Keyword
+
+- Default access modifier means we do not explicitly declare an access modifier for a class, field, method, etc.
+
+- A variable or method declared without any access control modifier is available to any other class in the same package. The fields in an interface
+  are implicitly public static final and the methods in an interface are by default public.
```

Comment:
I would remove the explanation about fields in an interface for now, because (1) we didn't show them interfaces yet and (2) it might be too much detail